src/scatter.jl

```diff
@@ -0,0 +1,40 @@
+for op in [+, -, *, /, max, min, &, |]
+    @eval function scatter_kernel!(op::typeof($(op)), dst, src, idx)
+        index = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+
+        @inbounds if index <= length(idx)
+            @atomic dst[idx[index]...] = $(op)(dst[idx[index]...], src[index])
+        end
+        return nothing
+    end
+
+    @eval function scatter_kernel!(op::typeof($(op)), dst, src, idx, dims::Val{N}, max_idx, max_dims_idx, dims_size) where {N}
+        index = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+
+        @inbounds if index <= max_idx
+            j, k = divrem(index-1, max_dims_idx)
+            dims_i = CartesianIndices(dims_size)[k+1]
+            @atomic dst[Tuple(dims_i)..., idx[j+1]...] = $(op)(dst[Tuple(dims_i)..., idx[j+1]...], src[index])
+        end
+        return nothing
+    end
+
+    @eval function NNlib.scatter!(op::typeof($(op)), dst::CuArray{Tdst}, src::CuArray{Tsrc}, idx::CuArray{<:IntOrIntTuple}, dims::Val{N}) where {Tdst,Tsrc,N}
```

Comment:
If you want to support wrapped arrays, it's best to use `AnyCuArray` here.

Reply:
Also, do you need the `eval` here? Can't you just define this for all functions (and have `@cuda` throw a method error if using an unsupported op)?

Reply:
Done.